x/leverage/keeper/collateral.go

```diff
@@ -7,10 +7,10 @@ import (
 	"github.com/umee-network/umee/v2/x/leverage/types"
 )
 
-// burnCollateral removes some uTokens from an account's collateral and burns them. This occurs
-// during liquidations.
-func (k Keeper) burnCollateral(ctx sdk.Context, addr sdk.AccAddress, coin sdk.Coin) error {
-	err := k.setCollateralAmount(ctx, addr, k.GetCollateralAmount(ctx, addr, coin.Denom).Sub(coin))
+// redeemCollateral decollateralize uTokens from the owner and sends the underlying tokens
+// by to the recipient. This occurs during liquidations.
+func (k Keeper) redeemCollateral(ctx sdk.Context, owner, recipient sdk.AccAddress, coin sdk.Coin) error {
```

Comment:
This proposed change is not the intended behavior:

Former effects of `burnCollateral`:
- Burns uTokens in module
- Reduces uToken supply (because they were burned)

Proposed effects:
- Sends some uTokens from module to `toAddr` using proposed `decollateralize` and reduces their collateral
- Burns someone else's uTokens from the module
- Reduces uToken supply (because they were burned)

Reply:
I've moved one instruction to the  `Liquidate` function. Note, that `redeemCollateral` is called only in `Liquidate`.

Motivation for this change was following:
+ `burnCollateral` doesn't have much context - we never burn tokens without doing something else.
+ `redeemCollateral` has a full use case: redeem uToken -> token, and destroy used uToken.

Reply:
I've fixed fixed `burnCollateral / redeemCollateral` and renamed it to `liquidateCollateral`